wrappers/web_socket.js

```
@@ -0,0 +1,303 @@
+      // Returns the length of the queue.
+      this.getLength = function(){
+          return (queue.length - offset);
+      };
+      
+      // Returns true if the queue is empty, and false otherwise.
+      this.isEmpty = function(){
+          return (queue.length == 0);
+      };
+      
+      /* Enqueues the specified item. The parameter is:
+       *
+       * item - the item to enqueue
+       */
+      this.enqueue = function(item){
+          queue.push(item);
+      };
+      
+      /* Dequeues an item and returns it. If the queue is empty, the value
+       * 'undefined' is returned.
+       */
+      this.dequeue = function(){
+          
+          // if the queue is empty, return immediately
+          if (queue.length == 0) return undefined;
+          
+          // store the item at the front of the queue
+          var item = queue[offset];
+          
+          // increment the offset and remove the free space if necessary
+          if (++ offset * 2 >= queue.length){
+              queue  = queue.slice(offset);
+              offset = 0;
+          }
+          
+          // return the dequeued item
+          return item;
+          
+      };
+      
+      /* Returns the item at the front of the queue (without dequeuing it). If the
+       * queue is empty then undefined is returned.
+       */
+      this.peek = function(){
+          return (queue.length > 0 ? queue[offset] : undefined);
+      };
+  };
+
+  var WebSocket_Create = function(instance) {
+    return resources.register(WEB_SOCKET_RESOURCE, {
+      destroy: function() {
+      }
+    });
+  };
+
+  var WebSocket_IsWebSocket = function(resource) {
+    return resources.is(resource, WEB_SOCKET_RESOURCE);
+  };
+  function performCallback(socket, data){
+      
+  }
+  var WebSocket_Connect = function(socket, url, protocols, protocol_count, callback) {
+    if (protocol_count > 1) {
+        // currently don't support protocol lists
+        return ppapi.PP_ERROR_FAILED;
```

Not 100% sure, but it may be better to throw an exception when you hit implementation flaws like this.  Inform the developer.